extensions/ANGLE_instanced_arrays/extension.xml

```diff
@@ -27,7 +27,7 @@
     </p>
   </overview>
   <idl xml:space="preserve">
-[NoInterfaceObject]
+[LegacyNoInterfaceObject]
```

Comment:
For this and other WebGL extensions, do you have any guidance about how to change the spec text to avoid the use of [LegacyNoInterfaceObject], but achieve the same goal - allow just type "object" to be returned from WebGLRenderingContextBase.getExtension(), not have the extensions' interface names appear in the global namespace, but have those objects include those interfaces? Would interface mixins work?

Reply:
I'm not quite sure what the intent is here. `[LegacyNoInterfaceObject]` doesn't change the type returned; this is still a proper class, named `ANGLE_instanced_arrays`. You can verify that using e.g. `ctx.getExtension("...").constructor.name === "ANGLE_instanced_arrays"`. (Not sure I got the syntax right for obtaining an instance, but you get the idea.)

The only thing `[LegacyNoInterfaceObject]` does is prevent `window.ANGLE_instanced_arrays` from existing. This, in turn, prevents feature detection (both of the overall feature, e.g. `if (window.ANGLE_instanced_arrays)`, and of individual methods, e.g. `if (window.ANGLE_instanced_arrays.drawArraysInstancedANGLE)`.

So my advice would be to just remove the `[LegacyNoInterfaceObject]` and allow the property to exist on the global. Hiding it doesn't seem to give much benefit. Maaaybe it'd be worth renaming them at the same time to something more idiomatic, e.g. `interface ANGLEInstancedArrays { ... }`, but I can see how having to define two names for things (one for use in the `getExtension()` argument and one for the global property) is not appealing, so having non-idiomatic naming for Web GL extension interfaces would be reasonable.

Reply:
It's arguably beneficial to prevent feature detection on `window.ANGLE_instanced_arrays`. Checking whether that class exists doesn't tell you anything about whether you can use it; it only tells you whether the browser implements the interface, so allowing it could be misleading. Feature detection must be done using getExtension() because support is hardware dependent.

Reply:
I disagree, because it makes up-front prototype patching impossible. We should preserve our existing behavior in this PR, and make a new issue for whether to change behavior.